src/uwtools/resources/jsonschema/cdeps.jsonschema

```diff
@@ -1,11 +1,92 @@
 {
+  "$defs": {
```

Comment:
I reworked the schema somewhat, and hope that I've captured the requirements correctly. When evaluating it, I'd suggest:
1. Reading through it to make sure it seems correct.
2. Noting that it successfully validates, as mentioned in a previous comment, the `docs/shared/cdeps.yaml` example file.
3. Looking at the new unit tests in `test_schemas.py`, below, which demonstrates many things that pass and fail when validated with this schema, or with parts of it.

It may also help to just read the file as-is, without considering the diff against the previous version.